core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt

```diff
@@ -295,57 +335,60 @@ object GraphQLActionBuilder {
-        val kind2 = elementInArgs?.type?.ofType?.ofType?.kind
-        if (kind0 == LIST) {//optional list in the top
+
+        val k = KindX()
+        val kindsLevels = k.quadKindsInInputs(elementInArgs)
+
+        if (kindsLevels.kind0 == LIST) {//optional list in the top
             inputElement.kindOfTableField = LIST
             inputElement.isKindOfTableFieldOptional = true
-            if (kind1 == NON_NULL) {// non optional input object or scalar
-                if (kind2 == INPUT_OBJECT || kind2 == SCALAR || kind2 == ENUM) {
-                    inputElement.kindOfTableFieldType = kind2
+            if (kindsLevels.kind1 == NON_NULL) {// non optional input object or scalar
+                if (kindsLevels.kind2?.let { kindChecksInput(it) }!!) {
+                    inputElement.kindOfTableFieldType = kindsLevels.kind2
                     inputElement.isKindOfTableFieldTypeOptional = false
-                    inputElement.tableFieldType = elementInArgs?.type?.ofType?.ofType?.name
-                    inputElement.tableField = elementInArgs?.name
+                    inputElement.tableFieldType = elementInArgs.type.ofType.ofType.name
+                    inputElement.tableField = elementInArgs.name
                     state.argsTables.add(inputElement)
                 }
             } else //optional input object or scalar or enum
-                if (kind1 == INPUT_OBJECT || kind1 == SCALAR || kind1 == ENUM) {
-                    inputElement.kindOfTableFieldType = kind1
+                if (kindsLevels.kind1?.let { kindChecksInput(it) }!!) {
+                    inputElement.kindOfTableFieldType = kindsLevels.kind1
                     inputElement.isKindOfTableFieldTypeOptional = true
-                    inputElement.tableFieldType = elementInArgs?.type?.ofType?.name
-                    inputElement.tableField = elementInArgs?.name
+                    inputElement.tableFieldType = elementInArgs.type.ofType.name
+                    inputElement.tableField = elementInArgs.name
                     state.argsTables.add(inputElement)
                 }
         } else // optional input object or scalar or enum in the top
-            if (kind0 == INPUT_OBJECT || kind0 == SCALAR || kind0 == ENUM) {
-                inputElement.kindOfTableFieldType = kind0
+            if (kindsLevels.kind0?.let { kindChecksInput(it) }!!) {
+                inputElement.kindOfTableFieldType = kindsLevels.kind0
                 inputElement.isKindOfTableFieldTypeOptional = true
-                inputElement.tableFieldType = elementInArgs?.type?.name
-                inputElement.tableField = elementInArgs?.name
+                inputElement.tableFieldType = elementInArgs.type.name
+                inputElement.tableField = elementInArgs.name
                 state.argsTables.add(inputElement)
             }
     }
 
+    private fun kindChecksInput(kind: __TypeKind) = kind == INPUT_OBJECT || kind == SCALAR || kind == ENUM
```

Comment:
same as before... need meanigful name (although this one was not too bad)
what about `isAnInputKind`
but then, is it exhaustive? for example, can a LIST not be an input type?